numba/unicode.py

```diff
@@ -540,6 +548,40 @@ def find_impl(a, b):
         return find_impl
 
 
+@overload_method(types.UnicodeType, 'count')
+def unicode_count(src, sub, start=None, end=None):
+
+    count_args_types_check(start)
+    count_args_types_check(end)
+
+    if isinstance(sub, types.UnicodeType):
```

Comment:
Can `sub` be other type like `types.UnicodeCharSeq` (Optional, None)?

Reply:
As I understand, it should be UnicodeType, due to UnicodeCharSeq does not match/cast cprreclty to unicode string (as mentioned in there: https://github.com/numba/numba/issues/4018)